### PR TITLE
feat(curriculum): create javascript variables and data types quiz

### LINKED ISSUE #56196: Variables and Data Types

### PR DESCRIPTION
**Pull request by spartanns**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #56196

### CODE REVIEW

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-variables-and-data-types/66edc25ae5ea80bf6f785552.md [hunk 1]
```diff
@@ -17,439 +17,497 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 2
+`const` is block scoped, `let` is not.
 
 ---
 
-Placeholder distractor 3
+`const` can be updated but not re-declared.
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-variables-and-data-types/66edc25ae5ea80bf6f785552.md:120]: imo, this is missing something about `let`

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-variables-and-data-types/66edc25ae5ea80bf6f785552.md [hunk 2]
```diff
@@ -17,439 +17,497 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --answer--
 
-Placeholder answer
+`const` cannot be updated or re-declared.
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-variables-and-data-types/66edc25ae5ea80bf6f785552.md:124]: same here, something about `let` is missing

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-variables-and-data-types/66edc25ae5ea80bf6f785552.md [hunk 3]
```diff
@@ -17,439 +17,497 @@ Answer all of the questions below correctly to pass the quiz.
...
+What does the code below do?
+
+```js
+let hello = "Hello";
+hello += " world";
+console.log(hello);
+```
+
 
 #### --distractors--
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-variables-and-data-types/66edc25ae5ea80bf6f785552.md:183]:
```suggestion

#### --distractors--
```

too many empty lines

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-variables-and-data-types/66edc25ae5ea80bf6f785552.md [hunk 4]
```diff
@@ -17,439 +17,496 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 2
+`const` is block scoped, `let` is not.
 
 ---
 
-Placeholder distractor 3
+`const` can be updated but not re-declared. `let` can be re-declared, but not updated.
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-variables-and-data-types/66edc25ae5ea80bf6f785552.md:120]:
```suggestion
`const` can be re-assigned but not re-declared. `let` can be re-declared, but not re-assigned.
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-variables-and-data-types/66edc25ae5ea80bf6f785552.md [hunk 5]
```diff
@@ -17,439 +17,496 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --answer--
 
-Placeholder answer
+`const` cannot be updated or re-declared. `let` can be updated, but not re-declared.
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-variables-and-data-types/66edc25ae5ea80bf6f785552.md:124]:
```suggestion
`const` cannot be re-assigned or re-declared. `let` can be re-assigned, but not re-declared.
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-variables-and-data-types/66edc25ae5ea80bf6f785552.md [hunk 6]
```diff
@@ -17,439 +17,496 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 2
+Variables declared in global scope can only be accessed from inside a function.
 
 ---
 
-Placeholder distractor 3
+Variables declared in global scope cannot be re-declared;
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-variables-and-data-types/66edc25ae5ea80bf6f785552.md:142]:
```suggestion
Variables declared in global scope cannot be re-declared.
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-variables-and-data-types/66edc25ae5ea80bf6f785552.md [hunk 7]
```diff
@@ -17,439 +17,496 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+You don't need to declare the data type of a variable before using it. JavaScript infers the type based on the value assigned to the variable.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of the following prints out the type of the variable x in the console?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-variables-and-data-types/66edc25ae5ea80bf6f785552.md:268]:
```suggestion
Which of the following prints out the type of the variable `x` in the console?
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
Off to a great start! Couple of small fixes.

**Review by naomi-lgbt — APPROVED** [review]
This looks good to go! Nice work~!

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-variables-and-data-types/66edc25ae5ea80bf6f785552.md [hunk 8]
```diff
@@ -181,15 +181,15 @@ console.log(hello);
 
 #### --distractors--
 
-Reassigns ` world` to the variable hello, and prints it out in the console.
+Reassigns ` world` to the variable `hello`, and prints it out in the console.
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-variables-and-data-types/66edc25ae5ea80bf6f785552.md:184]:
```suggestion
Reassigns `world` to the variable `hello`, and prints it out in the console.
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-variables-and-data-types/66edc25ae5ea80bf6f785552.md [hunk 9]
```diff
@@ -181,15 +181,15 @@ console.log(hello);
...
 
 ---
 
 Prints out the number of characters in `Hello world` in the console.
 
 ---
 
-Adds the string ` world` to the variable hello, and prints it out in the console.
+Adds the string ` world` to the variable `hello`, and prints it out in the console.
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-variables-and-data-types/66edc25ae5ea80bf6f785552.md:192]:
```suggestion
Adds the string `world` to the variable `hello`, and prints it out in the console.
```

**Review by huyenltnguyen — APPROVED** [review]
@spartanns Thank you for the PR, and congrats on your first contribution to the freeCodeCamp repo 🎉 

Happy contributing!

**Pull request merged**